### PR TITLE
testutil/beaconmock: initial PoC for beacon fuzz

### PR DESCRIPTION
**Pull request by dB2510**
Adds initial PoC for beacon fuzz to fuzz test charon where beacon mock returns random/malicious responses to charon.

category: feature
ticket: #1848

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -144,6 +144,7 @@ Flags:
       --p2p-tcp-address strings            Comma-separated list of listening TCP addresses (ip and port) for libP2P traffic. Empty default doesn't bind to local port therefore only supports outgoing connections.
       --private-key-file string            The path to the charon enr private key file. (default ".charon/charon-enr-private-key")
       --simnet-beacon-mock                 Enables an internal mock beacon node for running a simnet.
+      --simnet-beaconmock-fuzz             Configures simnet beaconmock to return fuzzed responses.
```
Comment (corverroos) [inline docs/configuration.md:147]: spelling, see above `simnet-beacon-mock`

**Review by corverroos — APPROVED** [review]

File: app/app.go [hunk 2]
```diff
@@ -79,6 +79,7 @@ type Config struct {
 	SimnetSlotDuration      time.Duration
 	SyntheticBlockProposals bool
 	BuilderAPI              bool
+	SimnetBeaconmockFuzz    bool
```
Comment (corverroos) [inline app/app.go:82]: nit: `SimnetBMockFuzz`

File: docs/configuration.md (hunk 1 above)
Comment (dB2510) [reply docs/configuration.md]: done

File: app/app.go (hunk 2 above)
Comment (dB2510) [reply app/app.go]: done

**Pull request merged**